Utility/DynamicJsonParser.cs

```diff
@@ -135,6 +137,73 @@ private static dynamic GetDynamic(PSObject obj, string numericalPrefix = "n", bo
                     propertyName = propertyName.ToLowerInvariant();
                 }
 
+                // Nuspec contains xml metadata as a string (only applicable to VSTS feeds)
+                // Example nuspec property:
+                // "nuspec":"<?xml version=\"1.0\" encoding=\"utf-8\"?>
+                //           <package xmlns=\"http://schemas.microsoft.com/packaging/2011/08/nuspec.xsd\">
+                //                <metadata>
+                //                      <id>Microsoft.Kusto.Tools</id>
+                //                      <version>1.0.6</version>
+                //                      <title>Kusto Tools</title>
+                //                      <authors>Microsoft</authors>
+                //                      <owners>Microsoft</owners>
+                //                      <requireLicenseAcceptance>false</requireLicenseAcceptance>
+                //                      <licenseUrl>http://kusto.blob.core.windows.net/kusto-nuget/EULA-agreement.htm</licenseUrl>
+                //                      <projectUrl>http://aka.ms/kdocs</projectUrl>
+                //                      <iconUrl>http://go.microsoft.com/fwlink/?LinkID=288890</iconUrl>
+                //                      <description>Kusto Tools</description>
+                //                      <releaseNotes>1.0.6: LightIngest refactored</releaseNotes>
+                //                      <copyright>Copyright ©  Microsoft Corporation</copyright>
+                //              </metadata>
+                //          </package>"
+                if (string.Equals(propertyName, "nuspec", StringComparison.OrdinalIgnoreCase))
+                {
+                    string xmlString = (string)psProperty.Value;
+
+                    if (string.IsNullOrWhiteSpace(xmlString))
+                    {
+                         var message = string.Format(Messages.InvalidNuspec, "xmlString");
+                         throw new InvalidDataException(message);
+                    }
+
+                    // Remove any characters that may come before the xml tag
+                    // Characters may appear before the start of an xml tag if xml encoding is specified
+                    if ((xmlString[0] != '<') && xmlString.Contains('<'))
+                    {
+                        var strIndex = xmlString.IndexOf('<');
+                        xmlString = xmlString.Substring(strIndex, xmlString.Length - 1);
```

Comment:
always 1?

Reply:
I'm assuming the nuspec content is valid from '<' on since what's contained the in the nuspec file can vary.

Reply:
That's possibly the reason why you are getting indexoutofbound for string like "???".  Should we  `xmlString = xmlString.Substring(strIndex, xmlString.Length - strIndex` instead?

Reply:
The index out of bounds error only happens for cases where '<' doesn't exist in the string.  I have substring(strIndex, xmlString.Length - 1)  so that if I have a situation such as  "?<expected nuspec content>"  (which is what I was getting when testing with Kusto Tools because of a weird thing that happens when xml encoding is specified in the xml tag) I can just remove the unwanted '?' that appears before the expected nuspec content.